File: 225.rb
# http://codegolf.stackexchange.com/questions/28786/write-a-program-that-makes-2-2-5

two_plus_two = "2 + 2"

class Fixnum
  alias :plus :+

  def +(value)
    self.plus(value).plus(1)
  end

end		

puts two_plus_two + " = " + eval(two_plus_two).to_s

File: 225.swift
// http://codegolf.stackexchange.com/questions/28786/write-a-program-that-makes-2-2-5

func +(left: Int, right: Int) -> Int {
  return (left == 2 && right == 2) ? add(add(left,right),1) : add(left,right)
}

func add(one: Int, two: Int) -> Int {
  return two == 0 ? one : add(one.successor(), two.predecessor())
}

println(2+2)
